### PR TITLE
Modernizing Configuration, Regions, Relations, Religions, Wars, World

### PR DESCRIPTION
**Pull request by IhateTrains**

### CODE REVIEW

File: EU4toV2/Source/EU4World/Regions/Areas.cpp [hunk 1]
```diff
@@ -12,9 +12,9 @@ EU4::Areas::Areas(std::istream& theStream)
 
 void EU4::Areas::registerKeys()
 {
-	registerRegex(R"([\w_]+)", [this](const std::string& areaName, std::istream& theStream) {
+	registerRegex(commonItems::stringRegex, [this](const std::string& areaName, std::istream& theStream) {
```
Comment (Zemurin) [inline EU4toV2/Source/EU4World/Regions/Areas.cpp:15]: unless stringregex includes underscores, this isn't a replacement.

File: EU4toV2/Source/EU4World/Religions/ReligionGroup.cpp [hunk 2]
```diff
@@ -21,7 +21,7 @@ void EU4::ReligionGroup::registerKeys()
 	registerKeyword("harmonized_modifier", commonItems::ignoreItem);
 	registerKeyword("crusade_name", commonItems::ignoreItem);
 
-	registerRegex("[a-zA-Z0-9_]+", [this](const std::string& religionName, std::istream& theStream) {
+	registerRegex(commonItems::stringRegex, [this](const std::string& religionName, std::istream& theStream) {
```
Comment (Zemurin) [inline EU4toV2/Source/EU4World/Religions/ReligionGroup.cpp:24]: underscore.

File: EU4toV2/Source/EU4World/Religions/Religions.cpp [hunk 3]
```diff
@@ -30,7 +30,7 @@ EU4::Religions::Religions(std::istream& theStream)
 
 void EU4::Religions::registerKeys()
 {
-	registerRegex("[a-zA-Z0-9_]+", [this](const std::string& unused, std::istream& theStream) {
+	registerRegex(commonItems::stringRegex, [this](const std::string& unused, std::istream& theStream) {
```
Comment (Zemurin) [inline EU4toV2/Source/EU4World/Religions/Religions.cpp:33]: ditto.

**Review by Zemurin — CHANGES_REQUESTED** [review]

File: EU4toV2/Source/EU4World/Regions/Areas.cpp (hunk 1 above)
Comment (IhateTrains) [reply EU4toV2/Source/EU4World/Regions/Areas.cpp:15]: Doesn't it? Most of the regex is about what's **not** included. The test are green.

**Pull request merged**